modules/osm/way.js

```diff
@@ -138,31 +138,53 @@ Object.assign(osmWay.prototype, {
     },
 
 
-    isOneWay: function() {
-        // explicit oneway tag..
-        var values = {
-            'yes': true,
-            '1': true,
-            '-1': true,
-            'reversible': true,
-            'alternating': true,
-            'no': false,
-            '0': false
-        };
-        if (values[this.tags.oneway] !== undefined) {
-            return values[this.tags.oneway];
+    /** @returns {boolean} for example, if `oneway=yes` */
+    isOneWayForwards() {
+        if (this.tags.oneway === 'no' || this.tags.oneway === '0') return false;
+
+        for (const key in this.tags) {
+            const value = this.tags[key];
+            const match = osmOneWayForwardTags[key]?.[value];
+            if (match) {
+                return match;
+            }
+        }
+        return false;
+    },
+
+    /** @returns {boolean} for example, if `oneway=-1` */
+    isOneWayBackwards() {
+        if (this.tags.oneway === 'no' || this.tags.oneway === '0') return false;
```

Comment:
As `oneway=0` is pretty much [obsolete](https://taghistory.raifer.tech/?#***/oneway/0) nowadays, we could perhaps simplify this to:

```js
        if (this.tags.oneway === 'no') return false;
```